lib/components/viewers/view-trip-button.tsx

```diff
@@ -0,0 +1,64 @@
+import { Button } from 'react-bootstrap'
+import { connect } from 'react-redux'
+import { FormattedMessage } from 'react-intl'
+import React, { Component } from 'react'
+
+import {
+  MainPanelContent,
+  setMainPanelContent,
+  setViewedTrip
+} from '../../actions/ui'
+
+type Props = {
+  fromIndex: number
+  setMainPanelContent: (panel: number | null) => void
+  setViewedTrip({
+    fromIndex,
+    toIndex,
+    tripId
+  }: {
+    fromIndex: number
+    toIndex: number
+    tripId: string
+  }): void
+  text: JSX.Element
+  toIndex: number
+  tripId: string
+}
+
+class ViewTripButton extends Component<Props> {
+  _onClick = () => {
+    this.props.setMainPanelContent(MainPanelContent.TRIP_VIEWER)
+    this.props.setViewedTrip({
```

Comment:
Can `setViewedTrip` also set the main panel content?

Reply:
See previous comments about avoiding regressions. I've mentioned elsewhere that I think our entire routing system needs a major overhaul.